src/components/ConsoleComponent.js

```diff
@@ -59,6 +59,16 @@ const useStyles = makeStyles((theme) => ({
 export default function Console(props) {
   const classes = useStyles();
   const [problemsView, setProblemsView] = useState(false);
+  const { problemCount, setExpandConsole } = props;
+
+  useEffect(() => {
+    // When props.problemCount changes and if there is a problem, make sure to open the console.
+    // If the problemCount is 0, we don't need to open it.
+    // The console should be able to be closed, so don't track whether the props.expandConsole value has changed.
+    if (problemCount) {
+      setExpandConsole(true);
```

Comment:
Do you think we should set the console to open to the problems view when this happens? On the one hand, I think that extra bit of red conveys that something went wrong here. On the other, there may be some messages logged to the main console that may help to diagnose the problem further.

Reply:
That's a good question. If we did add that functionality, does that mean it should swap over to the problems Problems tab when there is an error but the console was already open to the full Console tab? I think the way it is now means that whichever tab you already have open will stay open any time you use one of the "convert" buttons if the console is open. It feels a little odd to swap to the Problems tab if the Console tab is open and you are looking for messages there.

I'm thinking that with that piece and the fact that you do get the additional context, maybe it's not worth adding in. But also open to your opinion too!

Reply:
Yeah I hadn't thought about that, having the console switch to the other view if it's already open would feel really strange. I think this is the best way to handle that!

Reply:
How hard would it be to open to the problems tab if the console is collapsed, but leave it as-is if the console is already expanded?  If not difficult, what do you think about that approach?

Reply:
I was just looking into that. I'll report back.

Reply:
I remembered why this was harder than I thought. At least the way I have it set up with the `useEffect`, there isn't a way to tell if the console is closed because it was already closed and it should be opening on the "convert" buttons running, or if it's closed because we just clicked the collapse button and we want it to stay closed. There might be a way to move where the function is called, but I'm not sure what the right place for it should be.

Reply:
The only other place I can think to put this functionality is within the function that sets the messages themselves. But since that is just attached to the `console.log` function we overwrite, it isn't really tied to any of the state functions we use to open and close the console. I might be over-complicating it, but I can't figure out how to piece those two things together. If we ever decide to color code the console messages (like we get in a terminal for SUSHI), we might need to update how that function works, and maybe we can update this functionality then.

But also, if I'm over complicating things and either of you see an easy solution I'm missing, please let me know!